pkg/ottl/contexts/internal/ctxprofilecommon/attributes_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ctxprofilecommon // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/internal/ctxprofilecommon"
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+)
+
+// Mock implementations for AttributeContext and dependencies
+
+type mockAttributeContext struct {
+	indices    pcommon.Int32Slice
+	dictionary pprofile.ProfilesDictionary
+}
+
+func (m *mockAttributeContext) AttributeIndices() pcommon.Int32Slice {
+	return m.indices
+}
+
+func (m *mockAttributeContext) GetProfilesDictionary() pprofile.ProfilesDictionary {
+	return m.dictionary
+}
+
+func TestAccessAttributes_Getter(t *testing.T) {
```

Comment:
I'm know a few e2e tests are indirectly testing it already, but it would be nice to cover it with unit tests as well, witch are closer to the implementation and would give us extra confidence on this. 

A few examples we could test:
- Get all attributes (covered)
- Set all attributes (asserting it removes existing attributes indexes, keeping the dict table values, add the new values to the dict table & update attributes indexes references).
- Get attribute key
- Set attribute key (copy old attribute value, add new entry to the dict table, update attribute index, ensure old value isn't changed).

WDYT?

Reply:
> As the global AttributeTable in ProfilesDictionary is changed, it will effect all message Profile/Sample/Location and any other kind of message that references these attributes.

AFAIK, the global `AttributeTable` values in `ProfilesDictionary` should not be changed, instead, it should create a new entry and update the referrer's indices. There was some discussions about that on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40931#discussion_r2194394379.

With that in mind, what I meant here was doing something similar to the profile's e2e [dictionary validator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/e2e/profiles/e2e_test.go#L341).